Classes/Hooks/FileDumpHook.php

```diff
@@ -237,6 +237,9 @@ protected function dumpFileContents($file, $asDownload, $resumableDownload)
 
         // Find part of file and push this out
         $filePointer = @fopen($file->getForLocalProcessing(false), 'rb');
+        if ($filePointer === false) {
```

Comment:
Tnx for your PR

Can you move this above line 222 and add a `header('HTTP/1.1 404 File not found');` before the `exit;`?